src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -150,7 +150,7 @@ public async Task<IReadOnlyList<ResourceWrapper>> GetAsync(IReadOnlyList<Resourc
 
                 if (filtered.Any())
                 {
-                    await MergeInternalAsync(filtered, results, 1, cancellationToken);
+                    await MergeInternalAsync(filtered, results, maxParallelism: 1, cancellationToken);
                 }
             }
```

Comment:
If Max=1 so start with, then you probably don't need this whole block since its just retrying rateexceeded

```
 KeyValuePair<DataStoreOperationIdentifier, DataStoreOperationOutcome>[] retrySequentially =
                results.Where(x => x.Value.Exception is RequestRateExceededException).ToArray();
            if (retrySequentially.Any())
            {
                ResourceWrapperOperation[] filtered = resources
                    .Where(x => retrySequentially.Any(y => x.GetIdentifier().Equals(y.Key)))
                    .ToArray();

                if (filtered.Any())
                {
                    await MergeInternalAsync(filtered, results, maxParallelism: 1, cancellationToken);
                }
            }
```

Reply:
Thanks for the review, but these changes are not ready yet. I made these changes to run a load test and validate the data layer running in sequential at this part of the code. I'll get back with more updates.